x/name/keeper/keeper.go

```diff
@@ -291,3 +294,34 @@ func isValidUUID(s string) bool {
 	}
 	return true
 }
+
+func (keeper Keeper) addRecord(ctx sdk.Context, name string, addr sdk.AccAddress, restrict, isModifiable bool) error {
+	key, err := types.GetNameKeyPrefix(name)
+	if err != nil {
+		return err
+	}
+
+	store := ctx.KVStore(keeper.storeKey)
+	if store.Has(key) && !isModifiable {
+		return types.ErrNameAlreadyBound
+	}
+
+	record := types.NewNameRecord(name, addr, restrict)
+	if err = record.ValidateBasic(); err != nil {
```

Comment:
I'm wondering if we should start renaming these to just `Validate`.   Since, `ValidateBasic` is supposed to be for sdk.Msg types?

Reply:
While it'd be nice to do that, it's probably better left for a PR doing only that for any applicable `ValidateBasic` functions.